### PR TITLE
889 TC- WalkerColina -02

### PR DESCRIPTION
**Pull request by wcolina**
Test case #41 in the excel file.

**Test case:**
image

**Automated test case result:**
image

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationCreate.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    private WebElement nameTextInput;
+
+    @FindBy(id = "txtPpeClassificationDescription")
+    private WebElement descriptionTextInput;
+
+    @FindBy(id = "btnPpeClassificationSave")
+    private WebElement saveButton;
+
+    public PPEClassificationCreate() {
+        CommonEvents.isPresent(saveButton);
```
Comment (airmiky) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationCreate.java]: usar CommonEvents.isVisible(saveButton);

File: src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationList.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    private WebElement newPPEClassificationButton;
+
+    @FindBy(className = "container")
+    private WebElement ppeClassificationContainer;
+
+    @FindBy(xpath = "//table[@id='tblPpeClassification']/descendant::tr[last()]/td[count(//table[@id='tblPpeClassification']/descendant::th[text()='Name'])]")
+    private WebElement lastPPEClassificationName;
+
+    public PPEClassificationList() {
+        CommonEvents.isPresent(ppeClassificationContainer);
```
Comment (airmiky) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationList.java]: Use CommonEvents.isVisible(ppeClassificationContainer);

File: src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationList.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        return CommonEvents.isPresent(newPPEClassificationButton);
+    }
+
+    public PPEClassificationCreate clickNewPPEClassification(){
+        CommonEvents.clickButton(newPPEClassificationButton);
+        return new PPEClassificationCreate();
+    }
+
+    public String getLastPPEClassificationNameInTable() {
+        return lastPPEClassificationName.getText();
```
Comment (airmiky) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationList.java]: Utilizar CommonEvents.getTextContent(lastPPEClassificationName);

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 4]
```diff
@@ -105,13 +113,51 @@ public void clickAccidentMenu() throws Throwable {
     public void isAccidentListPresent() throws Throwable {
         Assert.assertTrue(accidentList.isAccidentListPresent(), "Fail, Accident List is not loaded");
     }
-    // PPE List
+
```
Comment (airmiky) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: Eliminar espacio

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java [hunk 5]
```diff
@@ -105,13 +113,51 @@ public void clickAccidentMenu() throws Throwable {
...
+        ppeClassificationList = ppeClassificationCreate.clickSaveButton();
+    }
+
+    @Then("^Created 'PPE Classification' is showed in PPE Classification list page$")
+    public void createdPPEClassificationIsShowedInPPEClassificationListPage() throws Throwable {
+        Assert.assertEquals(ppeClassificationList.getLastPPEClassificationNameInTable(), DataDriverTest.readValues.getValue("PPEClassification.create.name"), "Fail, PPE Classification is not created");
+    }
+
+    // Existing PPE List
+
```
Comment (airmiky) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: Eliminar espacio

File: src/test/resources/features/ppe/PPEClassificationCreate.feature [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# @Author: Walker Colina
+# feature to review behaviour of create ppe classification
+Feature: PPEClassificationCreate
+
+  Scenario: La inserción de registro Clasificación de equipo de protección personal se muestra en la lista de Clasificación de equipo de protección personal cuando presiono botón Save
+    Given 'SSI-A' home page is loaded
+    And Click 'PPE' menu on 'Header' page
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationCreate.feature]: Al inicio utilizar minuscula "click"

File: src/test/resources/features/ppe/PPEClassificationCreate.feature [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    And Click 'PPE Classification' sub menu of 'PPE' menu
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationCreate.feature]: Al inicio utilizar minuscula "click"

File: src/test/resources/features/ppe/PPEClassificationCreate.feature [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    And Click 'New PPE Classification' button of 'PPE Classification list' page
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationCreate.feature]: Al inicio utilizar minuscula "click"

File: src/test/resources/features/ppe/PPEClassificationCreate.feature [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+    And Fill 'PPE Classification' form using Data Driver Test on create 'PPE Classification' page
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationCreate.feature]: Al inicio utilizar minuscula "fill"

File: src/test/resources/features/ppe/PPEClassificationCreate.feature [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+    And Click 'Save' button into create 'PPE Classification' form page
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationCreate.feature]: Al inicio utilizar minuscula "click"

File: src/test/resources/features/ppe/PPEClassificationCreate.feature [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+    Then Created 'PPE Classification' is showed in PPE Classification list page
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationCreate.feature]: Al inicio utilizar minuscula "created"
comillas simples en 'PPE Classification list'

File: src/test/resources/features/ppe/PPEClassificationList.feature [hunk 12]
```diff
@@ -0,0 +1,9 @@
+# @Author: Walker Colina
+# feature to review behaviour of ppe classification list
+Feature: PPEClassificationList
+
+  Scenario: Clasificacion de equipo de proteccion personal - Todos los datos de clasificacion de equipos de proteccion personal son mostrados en una lista
+    Given 'SSI-A' home page is loaded
+    And Click 'PPE' menu on 'Header' page
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationList.feature]: Al inicio utilizar minuscula "click"

File: src/test/resources/features/ppe/PPEClassificationList.feature [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+    And Click 'PPE Classification' sub menu of 'PPE' menu
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationList.feature]: Al inicio utilizar minuscula "click"

File: src/test/resources/features/ppe/PPEClassificationCreate.feature [hunk 14]
```diff
@@ -0,0 +1,12 @@
+# @Author: Walker Colina
+# feature to review behaviour of create ppe classification
+Feature: PPEClassificationCreate
+
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationCreate.feature:4]: Eliminar espacio

File: src/test/resources/features/ppe/PPEClassificationList.feature [hunk 15]
```diff
@@ -0,0 +1,9 @@
+# @Author: Walker Colina
+# feature to review behaviour of ppe classification list
+Feature: PPEClassificationList
+
```
Comment (airmiky) [inline src/test/resources/features/ppe/PPEClassificationList.feature:4]: Eliminar espacio

**Review by airmiky — COMMENTED** [review]
Revisar las observaciones por favor

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java (hunk 5 above)
Comment (wcolina) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: Deleted all comment.

File: src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationCreate.java (hunk 1 above)
Comment (wcolina) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationCreate.java]: Done.

File: src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationList.java (hunk 2 above)
Comment (wcolina) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationList.java]: Done.

File: src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationList.java (hunk 3 above)
Comment (wcolina) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/ppe/PPEClassificationList.java]: Done.

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java (hunk 4 above)
Comment (wcolina) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java]: Done.

File: src/test/resources/features/ppe/PPEClassificationCreate.feature (hunk 6 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationCreate.feature]: Done.

File: src/test/resources/features/ppe/PPEClassificationCreate.feature (hunk 7 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationCreate.feature]: Done.

File: src/test/resources/features/ppe/PPEClassificationCreate.feature (hunk 8 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationCreate.feature]: Done.

File: src/test/resources/features/ppe/PPEClassificationCreate.feature (hunk 9 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationCreate.feature]: Done.

File: src/test/resources/features/ppe/PPEClassificationCreate.feature (hunk 10 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationCreate.feature]: Done.

File: src/test/resources/features/ppe/PPEClassificationCreate.feature (hunk 11 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationCreate.feature]: Done.

File: src/test/resources/features/ppe/PPEClassificationList.feature (hunk 12 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationList.feature]: Done.

File: src/test/resources/features/ppe/PPEClassificationList.feature (hunk 13 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationList.feature]: Done.

File: src/test/resources/features/ppe/PPEClassificationCreate.feature (hunk 14 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationCreate.feature:4]: No changes here. The blank line is generated for the ''Reformat Code' option.

File: src/test/resources/features/ppe/PPEClassificationList.feature (hunk 15 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationList.feature:4]: No changes here. The blank line is generated for the ''Reformat Code' option.

**Comment by wcolina** [comment]
Fixed all mentioned in the feedback.

**Comment by wcolina** [comment]
Renamed test case title.

File: src/test/resources/features/ppe/PPEClassificationList.feature (hunk 15 above)
Comment (wcolina) [reply src/test/resources/features/ppe/PPEClassificationList.feature]: Restored chromedriver and geckodriver files deleted by error

**Comment by wcolina** [comment]
Restored chromedriver and geckodriver files deleted by error

**Pull request merged**